src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/services/DraftAdjudicationService.kt

```diff
@@ -57,27 +67,55 @@ class DraftAdjudicationService(val draftAdjudicationRepository: DraftAdjudicatio
     val draftAdjudication = draftAdjudicationRepository.findById(id).orElseThrow { throwEntityNotFoundException(id) }
 
     if (draftAdjudication.incidentDetails == null)
-      throw EntityNotFoundException("DraftAdjudication does not have any incident details to update")
+      throw EntityNotFoundException("DraftAdjudication does not include an incident statement")
 
     locationId?.let { draftAdjudication.incidentDetails?.locationId = it }
     dateTimeOfIncident?.let { draftAdjudication.incidentDetails?.dateTimeOfIncident = it }
 
-    return draftAdjudication.toDto()
+    return draftAdjudicationRepository
+      .save(draftAdjudication)
+      .toDto()
+  }
+
+  @Transactional
+  fun editIncidentStatement(id: Long, statement: String): DraftAdjudicationDto {
+    val draftAdjudication = draftAdjudicationRepository.findById(id).orElseThrow { throwEntityNotFoundException(id) }
+
+    if (draftAdjudication.incidentStatement == null)
+      throw EntityNotFoundException("DraftAdjudication does not have any incident statement to update")
+
+    if (authenticationFacade.currentUsername != draftAdjudication.incidentStatement?.createdByUserId)
```

Comment:
just checking... this is inconsistent with editing the details as that can be edited without these checks - is that intended.

Reply:
This is correct. We are allowing anyone to make changes to the incident details. This is because the reviewer will sometimes correct some of the details regarding the incident. However, only the original author can make changes to the incident statement.